helm/cluster-api/templates/crd-install/crd-job.yaml

```diff
@@ -0,0 +1,70 @@
+{{- if .Values.crds.install }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "cluster-api.crdInstall" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    # create hook dependencies in the right order
+    "helm.sh/hook-weight": "-1"
+    {{- include "cluster-api.CRDInstallAnnotations" . | nindent 4 }}
+  labels:
+    app.kubernetes.io/component: {{ include "cluster-api.crdInstall" . | quote }}
+    {{- include "cluster-api.selectorLabels" . | nindent 4 }}
+    role: {{ include "cluster-api.CRDInstallSelector" . | quote }}
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: {{ include "cluster-api.crdInstall" . | quote }}
+        {{- include "cluster-api.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "cluster-api.crdInstall" . }}
+      securityContext:
+        runAsUser: 65534
+        runAsGroup: 65534
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: kubectl
+        image: "{{ .Values.registry.domain }}/giantswarm/docker-kubectl:latest"
+        command:
+        - sh
+        - -c
+        - |
+          set -o errexit ; set -o xtrace ; set -o nounset
+
+          # piping stderr to stdout means kubectl's errors are surfaced
+          # in the pod's logs.
+
+          kubectl apply -k /files 2>&1
```

Comment:
Helm template for generating a Job that installs CRDs. We mount all kustomize CRD files and run `kubectl` with `-k` flag as we are applying kustomize templates.